lms/models/lti_user.py

```diff
@@ -42,6 +42,21 @@ def is_learner(self):
         """Whether this user is a learner."""
         return "learner" in self.roles.lower()
 
+    @staticmethod
+    def from_auth_params(application_instance, auth_params):
```

Comment:
auth_params is not just some random dict, is of the type LTIAuthParamsSchema.

I had to do a lot of ceremony to annotate it here. I'll revisit it later.
https://github.com/hypothesis/lms/pull/3804/commits/ab216ec02302bda69c401dca1a5544bd5737fd18

Reply:
This would be fine content for a doc-string. I think we should discuss turning on the linting requirement again to document things as I feel our default has changed from documenting most things with a few exceptions, to not documenting things by default.

It's annoying to write things like `Intialize the object` for init methods, but I think it's probably on balance better than not having docstrings for our methods.

Reply:
This seems to be duplicate code from: 

* https://github.com/hypothesis/lms/blob/jwt-auth/lms/validation/authentication/_launch_params.py#L67

We should try and collapse these two paths to be the same path. Presumably we can remove the testing from one or the other too? (If there's any)